src/libostree/ostree-sysroot.c

```diff
@@ -1738,6 +1743,26 @@ ostree_sysroot_simple_write_deployment (OstreeSysroot      *sysroot,
   if (!booted_deployment && !merge_deployment && (retain_pending || retain_rollback))
     retain = TRUE;
 
+  /* tracks current versioned deployment */
+  OstreeRepo *repo = ostree_sysroot_repo (sysroot);
+  const gchar *new_version =
+    _ostree_deployment_get_version (new_deployment, repo, error);
+
+  gboolean retained_previous_version = FALSE;
+  if (booted_deployment)
+    {
+      const gchar *booted_version =
+        _ostree_deployment_get_version (booted_deployment, repo, error);
+      retained_previous_version = (g_strcmp0 (booted_version, new_version) != 0);
+    }
+
+  if (!retained_previous_version && merge_deployment)
```

Comment:
Could also check here that the booted and merge are not the same, e.g. `&& !ostree_deployment_equal (merge_deployment, booted_deployment)`. But not a big deal either since we cache the version now (though actually, this will still do work in the `NULL` case -- we should probably have a `version_is_cached` boolean field too in `OstreeDeployment` or something, but we can do that as a follow-up!).

Reply:
Done